### PR TITLE
Remove TODOs for diagnostics that are semantic

### PR DESCRIPTION
**Pull request by ahoppen**
All of the removed TODOs are semantic and should be diagnosed in ASTGen.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/translated/UnknownPlatformTests.swift [hunk 1]
```diff
@@ -8,10 +8,7 @@ final class UnknownPlatformTests: XCTestCase {
       """
```
Comment (bnbarham) [inline Tests/SwiftParserTest/translated/UnknownPlatformTests.swift:8]: Is it worth having any of the tests in this file? It'd be good to check we parse this correctly, but the diagnostics would be tested in what will likely be a separate if config module.

Comment (ahoppen) [reply Tests/SwiftParserTest/translated/UnknownPlatformTests.swift:8]: I could add a few substructure tests once https://github.com/apple/swift-syntax/pull/868 is merged. 😉

Comment (bnbarham) [reply Tests/SwiftParserTest/translated/UnknownPlatformTests.swift:8]: Removing both these files and having a structural test for `#if someCall(arg) && def || def2` is what I'd have for the parser test. Or break that down into a bunch 🤷.

Comment (ahoppen) [reply Tests/SwiftParserTest/translated/UnknownPlatformTests.swift]: Decided to move them to IfconfigExprTests but keep them because why not.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**